tox.ini

```diff
@@ -1,44 +1,67 @@
 [tox]
 minversion = 3.7.0
 envlist =
+    build-docker
+    doc
+    format-check
+    format-enforce
     lint
+    metadata-validation
     py{27,35,36,37}-ansible{25,26,27}-{functional,unit}
-    format-check
-    doc
 skipdist = True
 skip_missing_interpreters = True
 
 [testenv]
-usedevelop = True
+
 passenv = *
 setenv =
     PYTHONDONTWRITEBYTECODE=1
 deps =
-    -rrequirements.txt
-    -rtest-requirements.txt
     ansible24: ansible>=2.4,<2.5
     ansible25: ansible>=2.5,<2.6
     ansible26: ansible>=2.6,<2.7
     ansible27: ansible>=2.7,<2.8
     ansibledevel: git+https://github.com/ansible/ansible.git
+    build-docker: pbr
+    format: yapf>=0.25.0,<2
+    {functional,unit}: -rtest-requirements.txt
+    lint: -rlint-requirements.txt
+    metadata-validation: collective.checkdocs
+    metadata-validation: twine
 extras =
-    azure
-    docker
-    docs
-    ec2
-    gce
-    linode
-    lxc
-    openstack
-    vagrant
-    windows
+    {functional,unit}: azure
+    {functional,unit}: docker
+    doc: docs
+    {functional,unit}: ec2
+    {functional,unit}: gce
+    {functional,unit}: linode
+    {functional,unit}: lxc
+    {functional,unit}: openstack
+    {functional,unit}: vagrant
+    {functional,unit}: windows
 commands_pre =
     find {toxinidir} -type f -not -path '{toxinidir}/.tox/*' -path '*/__pycache__/*' -name '*.py[c|o]' -delete
 commands =
-    unit: pytest test/unit/ --cov={toxinidir}/molecule/ --no-cov-on-fail {posargs}
+    doc: python setup.py build_sphinx -n -W --builder=html
+    format-check: yapf -d -r molecule/ test/
+    format-enforce: yapf -i -r molecule/ test/
     functional: pytest test/functional/ {posargs}
     lint: flake8
     lint: yamllint -s test/ molecule/
+    metadata-validation: python -m setup checkdocs check --metadata --restructuredtext --strict --verbose
+    metadata-validation: twine check .tox/dist/*
```

Comment:
I think that once `deps` or `commands` for a testenv break out to multiple lines, a dedicated `testenv:section` should probably be used. I love all the collapsed one-line dep/command testenvs though.